packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeActions.ts

```diff
@@ -36,6 +45,61 @@ export class TreeActions {
       parentChildMap: new Map(),
       rootNode: { id: undefined, nodeData: undefined },
     };
+
+    this._nodeLoader = this.createNodeLoader();
+  }
+
+  private createNodeLoader() {
+    const subject = new Subject<LoadDebouncedNodesOptions>();
+    subject
+      .pipe(
+        groupBy((item) => item.parentId),
+        mergeMap((group) =>
+          group.pipe(
+            buffer(group.pipe(debounceTime(0))),
+            map((groupArr) => {
+              const last = groupArr.length - 1;
+              let returnIndex = last;
+
+              groupArr.forEach((member, index) => {
+                if (member.loadOptions.discardState) {
+                  returnIndex = index;
+                  return;
+                }
+                if (index === last && last === returnIndex) {
+                  return;
+                }
```

Comment:
what's this for?

Reply:
We don't want to resolve the promise of last member within the array or if a discard/reset was passed then we do.

Reply:
I was about to suggest simply using `index === returnIndex` check, but now I think the only appropriate way is to have two loops:
1. iterate through all members to find the one we're going to use,
2. iterate through all members and call `onComplete` on each of them except the one we got from step 1.

Without that, we may be not calling `onComplete` on some of the members. E.g. see what happens if you call `reloadTree({ state: "discard" })` many times.

Reply:
You're right changed to have two loops.